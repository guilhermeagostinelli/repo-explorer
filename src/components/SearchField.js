import React from "react";
import Label from "./Label";

const SearchField = ({ label, value, onChange }) => (
  <Label>
    {label}
    <input
      type="search"
      value={value}
      onChange={e => onChange(e.target.value)}
    />
  </Label>
);

export default SearchField;
