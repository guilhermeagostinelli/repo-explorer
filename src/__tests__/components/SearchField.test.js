import React from "react";
import { create } from "react-test-renderer";
import SearchField from "../../components/SearchField";
import "jest-styled-components";

describe("<SearchField />", () => {
  it("should match the snapshot", () => {
    const searchField = create(
      <SearchField label="Pesquisa:" value="" onChange={() => {}} />
    );
    expect(searchField.toJSON()).toMatchSnapshot();
  });

  it("should fire the callback function when the user types something in the search field", () => {
    const handleChange = jest.fn();
    const searchField = create(
      <SearchField label="Pesquisa:" value="" onChange={handleChange} />
    );
    const instance = searchField.root;
    const input = instance.findByType("input");
    input.props.onChange({ target: { value: "abc" } });
    expect(handleChange).toHaveBeenCalledTimes(1);
    expect(handleChange).toHaveBeenCalledWith("abc");
  });
});
