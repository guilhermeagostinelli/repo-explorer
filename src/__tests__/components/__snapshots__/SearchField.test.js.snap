// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchField /> should match the snapshot 1`] = `
.c0 > input,
.c0 > select {
  margin-left: 10px;
}

<label
  className="c0"
>
  Pesquisa:
  <input
    onChange={[Function]}
    type="search"
    value=""
  />
</label>
`;
